assets/js/toggle.js

```diff
@@ -0,0 +1,36 @@
+function toggleDarkMode() {
+    const DARK_CLASS = 'dark';
+
+    var body = document.querySelector("body");
+    if (body.classList.contains(DARK_CLASS)) {
+        setCookie('theme', 'light');
+        body.classList.remove(DARK_CLASS);
+    } else {
+        setCookie('theme', 'dark');
+        body.classList.add(DARK_CLASS);
+    }
+}
+
+function getCookie(name) {
+    var v = document.cookie.match('(^|;) ?' + name + '=([^;]*)(;|$)');
+    return v ? v[2] : null;
+}
+function setCookie(name, value, days) {
+    var d = new Date;
+    d.setTime(d.getTime() + 24*60*60*1000*days);
+    document.cookie = name + "=" + value + ";path=/;expires=" + d.toGMTString();
+}
+
+function deleteCookie(name) { setCookie(name, '', -1); }
```

Comment:
I think we are better of adding a separate file for this, `darkmode.js` so that we don't load it if we don't need to.

Reply:
Yeah! you're definitely right, I din't think about that! I'll put the script in the if block, so it won't get loaded if dark mode is disabled!